### ISSUE TITLE
Build failure in Xcode 9b4

### ISSUE THREAD

**Issue by mrh-is** [adamwaite/Validator#91]
Three UIKit classes are reported as not conforming to `ValidatableInterfaceElement`:

```
/Pods/Validator/Validator/Sources/UIKit+Validator/UISlider+Validator.swift:33:1: error: type 'UISlider' does not conform to protocol 'ValidatableInterfaceElement'
extension UISlider: ValidatableInterfaceElement {
^
/Pods/Validator/Validator/Sources/UIKit+Validator/ValidatableInterfaceElement.swift:141:17: note: candidate has non-matching type '<Self, R> (rule: R) -> ValidationResult'
    public func validate<R: ValidationRule>(rule r: R) -> ValidationResult where R.InputType == InputType {
                ^
/Pods/Validator/Validator/Sources/UIKit+Validator/UISlider+Validator.swift:33:1: note: do you want to add protocol stubs?
extension UISlider: ValidatableInterfaceElement {
^
```

```
/Pods/Validator/Validator/Sources/UIKit+Validator/UITextField+Validator.swift:32:1: error: type 'UITextField' does not conform to protocol 'ValidatableInterfaceElement'
extension UITextField: ValidatableInterfaceElement {
^
/Pods/Validator/Validator/Sources/UIKit+Validator/ValidatableInterfaceElement.swift:141:17: note: candidate has non-matching type '<Self, R> (rule: R) -> ValidationResult'
    public func validate<R: ValidationRule>(rule r: R) -> ValidationResult where R.InputType == InputType {
                ^
/Pods/Validator/Validator/Sources/UIKit+Validator/UITextField+Validator.swift:32:1: note: do you want to add protocol stubs?
extension UITextField: ValidatableInterfaceElement {
^
```

```
/Pods/Validator/Validator/Sources/UIKit+Validator/UITextView+Validator.swift:11:1: error: type 'UITextView' does not conform to protocol 'ValidatableInterfaceElement'
extension UITextView: ValidatableInterfaceElement {
^
/Pods/Validator/Validator/Sources/UIKit+Validator/ValidatableInterfaceElement.swift:141:17: note: candidate has non-matching type '<Self, R> (rule: R) -> ValidationResult'
    public func validate<R: ValidationRule>(rule r: R) -> ValidationResult where R.InputType == InputType {
                ^
/Pods/Validator/Validator/Sources/UIKit+Validator/UITextView+Validator.swift:11:1: note: do you want to add protocol stubs?
extension UITextView: ValidatableInterfaceElement {
^
```

Happens on both Swift 3.3 and 4.1. I think this worked on 9b2, but not on 9b3 or 9b4.

**Comment by tuan188** [comment]
It happens on Xcode 9.3 (9E145), Swift 4.1 too.

**Comment by konkontos** [comment]
Don't have enough time to explore / follow the full logic of the codebase, but removing the protocol stub and default implementation of:

`public func validate<R: ValidationRule>(rule r: R) -> ValidationResult where R.InputType == InputType`

from `ValidatableInterfaceElement.swift`

seems to solve the issue.

In the App I'm working on, eliminating this method causes no problems.

**Comment by PaulScala** [comment]
Thank you @konkontos this fix worked for me too! 

I was getting the same error as @mrh-is in Xcode 9.3.

**Comment by mrh-is** [comment]
#93 is an identical bug with more chatter, so I'll close this in favor of that one.
